### DISCUSSION TITLE
scikit-learn dependency issues

### DISCUSSION THREAD

**Discussion by sr1998** [Trusted-AI/adversarial-robustness-toolbox#1950] [Ideas]
Hi, is it possible for the library to support newer versions of scikit-learn? At the moment it supports scikit-learn<1.1.0,>=0.22.2  but I also tested part of ART with scikit-learn 1.1.2 and it works like a charm. The question is of course whether everything will keep working which you know better of course. 

I hope support for newer versions of scikit-learn, at least 1.1.2 for me, is possible in the near future. I look forward to any reply.

**Comment by beat-buesser** [comment]
Hi @sr1998 Yes, I think that's possible. I'll convert this discussion into a feature request issue.

**Comment by jamesmyatt** [comment]
What's the logic for having an upper bound on the scikit-learn version anyway? Is there an expectation of breaking changes?

**Reply by beat-buesser** [reply]
Hi @jamesmyatt It has happened in previous versions that new scikit-learn releases introduces breaking changes. ART has to access sometime internal attribtues of scikit-learn models for the purpose of adversarial machine learning. These internal attributes can of course change between releases without warnings from scikit-learn.
